tests/ui/default_numeric_fallback_i32.fixed

```diff
@@ -76,7 +76,7 @@ mod function_def {
     fn ret_i32() -> i32 {
         // Even though the output type is specified,
         // this unsuffixed literal is linted to reduce heuristics and keep codebase simple.
-        1_i32
+        1
     }
 
```

Comment:
Question: Does this make sense? There was a comment in the linked issue suggest against it, and I kinda agree with it, but I'm not sure. If the original logic was fine I can simply remove those additional `ExprKind::` checks and this would be a small fix~

Reply:
This definitely shouldn't be linted and the comment should be removed.